internal/controller/controllers/installenv_controller.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2020.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/internal/common"
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/openshift/assisted-service/internal/bminventory"
+	adiiov1alpha1 "github.com/openshift/assisted-service/internal/controller/api/v1alpha1"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/restapi/operations/installer"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// InstallEnvReconciler reconciles a InstallEnv object
+type InstallEnvReconciler struct {
+	client.Client
+	Log                      logrus.FieldLogger
+	Scheme                   *runtime.Scheme
+	Installer                bminventory.InstallerInternals
+	PullSecretUpdatesChannel chan event.GenericEvent
+}
+
+// +kubebuilder:rbac:groups=adi.io.my.domain,resources=installenvs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=adi.io.my.domain,resources=installenvs/status,verbs=get;update;patch
+
+func (r *InstallEnvReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+
+	installEnv := &adiiov1alpha1.InstallEnv{}
+	if err := r.Get(ctx, req.NamespacedName, installEnv); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	return r.create(ctx, installEnv)
+}
+
+func (r *InstallEnvReconciler) create(ctx context.Context, installEnv *adiiov1alpha1.InstallEnv) (ctrl.Result, error) {
```

Comment:
It may help readability to name this something like `createSQL` to make it clear that you're not trying to re-create the k8s resource. A comment of course would also help explaining what this does.

Reply:
Maybe generate? SQL is only part of it, eventually the result is URL to the generated ISO.